### ISSUE TITLE
Bump ring version to 0.17

### ISSUE THREAD

**Issue by DragonDev1906** [sigp/ethereum_hashing#5]
I'd like to use `ethereum_hashing` (and more specifically `tree_hash` and `ssz_types`) for an application running on the `x86_64-fortanix-unknown-sgx` target. Unfortunately, the `ring` dependency version 0.16.x does not compile with this target, this was added last week (see the [addition of the feature to make it work/compile](https://github.com/briansmith/ring/commit/9195b4a33f0d1e00b8e8fd27b3562a80f81b1046), [feature renaming](https://github.com/briansmith/ring/commit/7fa58cc7f78b090181d1d771e441cef57adc46d0) and a [comment mentioning this](https://github.com/briansmith/ring/issues/744#issuecomment-1756037111).

```
ring v0.16.20
└── ethereum_hashing v1.0.0-beta.2
    └── tree_hash v0.5.2
        └── ssz_types v0.5.4
```

As of now, [ring](https://github.com/briansmith/ring/commits/main) does not have a release containing this commit, but this will likely be added in the next version (0.17.6). Can we bump the ring dependency to `0.17` or any `0.17.x`? That would allow using the commit that contains this change and would allow using the newest ring version as soon as it is released (unless there also is a breaking change in ring).

**Comment by michaelsproul** [comment]
Yeah this sounds good. I'm also inclined to say we should downgrade ethereum_hashing's version from a v1.0.0-beta, because this will be a breaking release. We already did this for the other crates that have non 1.0 dependencies, and I think it's the correct approach per Rust crate guidelines.

TL;DR: please PR a ring update and then I'll bump the version to something that isn't 1.0

**Comment by michaelsproul** [comment]
New version 0.6.0 is out, thanks!

**Comment by michaelsproul** [comment]
hey @DragonDev1906 I've also released tree_hash and ssz_types 0.6 now

**Comment by DragonDev1906** [comment]
Awesome, thank you :heart:
Since ring has now released a version with the commit I've mentioned (17.6 and 17.7) I should be able to use these crates on `x86_64-fortanix-unknown-sgx`.
